### PR TITLE
ADOPTED: TGUI Crew Credits

### PR DESCRIPTION
**Pull request by DisturbHerb**
## About the PR 
Adopts #12339 with some minor changes and feedback. The preview images differ from the original PR's output, please refer to the original PR to compare.

image
image

## Why's this needed? 
tgui good :)

## Changelog 

```changelog
(u)glowbold
(+)Crew credits now use TGUI - make sure to report any bugs!
```

### CODE REVIEW

File: tgui/packages/tgui/interfaces/CrewCredits.tsx
Comment (DisturbHerb) [inline tgui/packages/tgui/interfaces/CrewCredits.tsx:54]: i'll be honest with you, we kinda need to have 2 maps in case some department inexplicably has multiple heads

File: tgui/packages/tgui/interfaces/CrewCredits.tsx
Comment (DisturbHerb) [inline tgui/packages/tgui/interfaces/CrewCredits.tsx:16]: > If using `group` as a label below, should probably name `title` or `label` instead on the individual `group` items in the data structure.

mordent please clarify this

File: tgui/packages/tgui/interfaces/CrewCredits.tsx
Comment (DisturbHerb) [inline tgui/packages/tgui/interfaces/CrewCredits.tsx:56]: > Prefer `filter` or `find` vs. two `map`s.

i'll be honest with you, we kinda need to have 2 maps (as far as i know) in case some department inexplicably has multiple heads

**Comment by DisturbHerb** [comment]
!merge_upstream

File: code/datums/gameticker.dm
Comment (ZeWaka) [inline code/datums/gameticker.dm:770]: 🦀🦀🦀🦀🦀🦀🦀🦀🦀

File: code/datums/crew_credits.dm
Comment (ZeWaka) [inline code/datums/crew_credits.dm:4]: 🤔

File: code/datums/crew_credits.dm
Comment (ZeWaka) [inline code/datums/crew_credits.dm:227]:
```suggestion
			role=pick("Tourist", "Musician", "Union Rep", "Board Member", "Regional Director", "Inspector", "Governor", "Diplomat")
```

File: tgui/packages/tgui/interfaces/CrewCredits.tsx
Comment (mordent-goonstation) [reply tgui/packages/tgui/interfaces/CrewCredits.tsx]: `data.groups` is the list, and each item in that list is a group. The name (title/label) of the group should be `title`, `label`, or `name` and not `group` as it is here. `group.group` is weird,

**Comment by DisturbHerb** [comment]
!merge_upstream

File: tgui/packages/tgui/interfaces/CrewCredits.tsx
Comment (mordent-goonstation) [inline tgui/packages/tgui/interfaces/CrewCredits.tsx:1]: Can we get a file license header in here? I know some are missing them, but I'm trying to get all of our tgui interfaces to have licenses headers. See a bunch of the other interfaces for examples. MIT or ISC are common and work fine for us, but you can pick any license that will allow us to use it.

**Review by mordent-goonstation — APPROVED** [review]
Approving, once a license header is added feel free to merge.

**Pull request merged**
